Automattic/woocommerce-payments
# Fix platform checkout tracks for the express payment button.

#### Changes proposed in this Pull Request

Once we added the WooPay Express Checkout button functionality, `wcpayTracks` requests were failing due to the use of `getConfig` and how we were making the necessary parameters available when showing the express checkout button. This PR addresses this by adding a filter to the value returned by `WC_Payments_Checkout->get_payment_fields_js_config()` method. We're also attaching that config to our express payment button js file (it's normally only attached to the `checkout.js` file). This ensures that we can use `getConfig( 'value' )` for one of our custom config values anywhere we're displaying the WooPay express checkout button.

This PR also adds two new track events: `platform_checkout_express_button_offered` and `platform_checkout_express_button_clicked`. These events will also include a `context` event property that says which screen they originated from - `product`, `cart`, or `checkout`.

#### Testing instructions

1. Apply this PR.
2. Make sure the Express Checkout feature flag is enabled from the latest Dev Tools.
3. Make sure the Express Checkout button is set to display on the Product, Cart, and Checkout pages from Payments > Settings > WooPay Customize > Show express checkouts on.
4. Visit a single product page.
5. In your browser's `Network` tab, ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_offered
tracksEventProp: {"context":"product"}
```
6. Click the WooPay express payment button and ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_clicked
tracksEventProp: {"context":"product"}
```
7. Add the product to your cart and visit the cart page.
8. In your browser's `Network` tab, ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_offered
tracksEventProp: {"context":"cart"}
```
9. Click the WooPay express payment button and ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_clicked
tracksEventProp: {"context":"cart"}
```
10. Visit the checkout page.
11. In your browser's `Network` tab, ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_offered
tracksEventProp: {"context":"checkout"}
```
12. Click the WooPay express payment button and ensure that a call was made to `admin-ajax.php` with this in the `Payload`:
```
...
tracksEventName: platform_checkout_express_button_clicked
tracksEventProp: {"context":"checkout"}
```


-------------------


**Post merge**
